### PR TITLE
CI Refactor

### PR DESCRIPTION
**Pull request by mchrusci**
- Running CI from Jenkinsfile
- Added missing comments in utils.groovy
- Corrected cleanup stage

### CODE REVIEW

File: .ci/jenkins/utils/utils.groovy [hunk 1]
```diff
@@ -17,6 +17,14 @@
 STAGES_STATUS_MAP = [:]
 
 def GetDockerEnvList(String projectName, String dockerContainerNamePrefix, String projectRoot = projectName) {
+    /**
+    * Generates map for method using configurationMaps.
```
Comment (lwencel) [inline .ci/jenkins/utils/utils.groovy]: This method generates configuration map list using dockerfiles available in dockerfiles directory

File: .ci/jenkins/Jenkinsfile [hunk 2]
```diff
@@ -76,16 +85,19 @@ def RunToxTests(configurationMaps) {
     UTILS.CreateStage("Run_tox_tests", runToxTestsMethod, configurationMaps)
 }
 
-def Cleanup(configurationMaps) {
-    Closure cleanupMethod = { configMap ->
+def Cleanup() {
+    stage("Cleanup"){
```
Comment (lwencel) [inline .ci/jenkins/Jenkinsfile]: Why do you remove closures? Currently you don't have configMap variable so this method will fail

File: .ci/jenkins/Jenkinsfile [hunk 3]
```diff
@@ -117,16 +128,31 @@ def Notify() {
...
-    timeout(activity: true, time: 60) {
-        def configurationMaps = UTILS.GetDockerEnvList(projectName, dockerContainerName, projectRoot)
-        BuildImage(configurationMaps)
-        RunDockerContainers(configurationMaps)
-        PrepareEnvironment(configurationMaps)
-        RunToxTests(configurationMaps)
-        Cleanup(configurationMaps)
-        Notify()
+def main(String projectName, String projectRoot, String dockerContainerName, String jenkins_github_credential_id, String repository_git_address) {
+    node('ci && onnx') {
```
Comment (ldurka) [inline .ci/jenkins/Jenkinsfile]: maybe the node definition should be in  line 158 i.e. if we would like to change labels
```
node('ci && onnx') {
   main(PROJECT_NAME, CI_ROOT, DOCKER_CONTAINER_NAME, JENKINS_GITHUB_CREDENTIAL_ID, REPOSITORY_GIT_ADDRESS)
}
```
what do you think

File: .ci/jenkins/Jenkinsfile (hunk 2 above)
Comment (mchrusci) [reply .ci/jenkins/Jenkinsfile]: Curiously enough it didn't fail. My original thought was that Cleanup should execute even if job fails to generate configuration maps, but I see in that case this method would surely fail.
I restored older Cleanup and added job build directory removal in finally brackets after Cleanup.

**Pull request merged**